apis/v1alpha2/clusternetworkpolicy_types.go

```diff
@@ -0,0 +1,482 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// All fields in this package are required unless Explicitly marked optional
+// +kubebuilder:validation:Required
+package v1alpha2
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ClusterNetworkPolicy is a cluster-wide network policy resource.
+//
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:shortName=cnp,scope=Cluster
+// +kubebuilder:printcolumn:name="Tier",type=string,JSONPath=".spec.tier"
+// +kubebuilder:printcolumn:name="Priority",type=string,JSONPath=".spec.priority"
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=".metadata.creationTimestamp"
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type ClusterNetworkPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	// Spec defines the desired behavior of ClusterNetworkPolicy.
+	Spec ClusterNetworkPolicySpec `json:"spec"`
+
+	// Status is the status to be reported by the implementation.
+	// +optional
+	Status ClusterNetworkPolicyStatus `json:"status,omitempty"`
+}
+
+// ClusterNetworkPolicyStatus defines the observed state of
+// ClusterNetworkPolicy.
+type ClusterNetworkPolicyStatus struct {
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions" patchStrategy:"merge" patchMergeKey:"type"`
+}
+
+// ClusterNetworkPolicySpec defines the desired state of ClusterNetworkPolicy.
+type ClusterNetworkPolicySpec struct {
+	// Tier is used as the top-level grouping for network policy prioritization.
+	//
+	// Policy tiers are evaluated in the following order:
+	//
+	// * Admin tier
+	// * NetworkPolicy tier
+	// * Baseline tier
+	// * Default tier
```

Comment:
Would prefer a different name for this if we reference it in this way. Calico's "default" tier is the tier that we put k8s network policy in.